lib/improver/spotdata/extract_data.py

```diff
@@ -0,0 +1,374 @@
+
+            llr = local_lapse_rate(cube, orography, node_list)
+            data[i_site] = llr[0]*altitude + llr[1]
+
+        return self.make_cube(cube, data, sites)
+
+    def model_level_temperature_lapse_rate(self, cube, sites, neighbours,
+                                           pressure_on_height_levels,
+                                           surface_pressure,
+                                           temperature_on_height_levels):
+        """
+        Lapse rate method based on potential temperature. Follows the work of
+        S.B. Vosper 2005 (Near-surface temperature variations over complex
+        terrain).
+
+        Args:
+        -----
+        cube : iris.cube.Cube
+            A cube of screen level temperatures at a single time.
+
+        sites/neighbours : See process() above.
+
+        pressure_on_height_levels : iris.cube.Cube
+            Pressure data at 50m above the surface at an equivalent time to the
+            cube of screen level temperatures.
+
+        surface_pressure : iris.cube.Cube
+            Cube of surface pressures at an equivalent time to the cube of
+            screen level temperatures.
+
+        temperature_on_height_levels : iris.cube.Cube
+            Temperature data at 50m above the surface at an equivalent time to
+            the cube of screen level temperatures.
+
+        Returns:
+        --------
+        Data extracted from the screen level temperature cube at spotdata sites
+        which has then been adjusted using a temperature lapse rate calculated
+        using multi-level data.
+
+        """
+        if not cube.name() == 'air_temperature':
+            raise ValueError('{} should only be used for adjusting '
+                             'temperatured. Cube of type {} is not '
+                             'suitable.'.format(self.method, cube.name()))
+
+        kappa = R_DRY_AIR/CP_DRY_AIR
+
+        data = np.empty(shape=(len(sites)))
+        for i_site in range(len(sites)):
+
+            i, j, dz = (neighbours['i'][i_site], neighbours['j'][i_site],
+                        neighbours['dz'][i_site])
+
+            # Use neighbour grid point value if vertical displacement == 0.
+            if isclose(dz, 0.):
+                data[i_site] = cube.data[i, j]
+                continue
+
+            t_upper = temperature_on_height_levels.data[i, j]
+            p_upper = pressure_on_height_levels.data[i, j]
+            t_surface = cube.data[i, j]
+            p_surface = surface_pressure.data[i, j]
+
+            p_grad = (p_upper - p_surface)/50.
+            p_site = p_surface + p_grad*dz
+
+            theta_upper = t_upper*(1.0E5/p_upper)**kappa
+            theta_surface = t_surface*(1.0E5/p_surface)**kappa
+            dthetadz = (theta_upper - theta_surface)/50.
+
+            if abs(dz) < 1.:
+                t1p5 = t_surface
+            else:
+                dz = min(abs(dz), 70.)*np.sign(dz)
```

Comment:
I have made all these values configurable. I have added them to the configuration stuff in spotdata (that is not in this pull request), but I have set defaults for key word arguments so the plugin will work without the config being set. The values used as defaults are UKPP default values, but it might be nice to do some verification experiments on how changing these values affect the performance, hence allowing them to be set in the configs.